server/src/files.rs

```diff
@@ -0,0 +1,137 @@
+use std::{fmt, path::PathBuf, time::Duration};
+
+use opendal::{services::S3, Operator};
+use tokio::fs::File;
+
+use crate::{appstate::AppState, config::Config, errors::AtomicServerResult};
+
+pub enum FileStore {
+    S3,
+    FS,
```

Comment:
Perhaps this would a good place to store config data? We can then guarantee that if instantiation of the FileStore has succeeded, we've got all the data we need.

We could have a `FileStore::init_from_config` that reads all the relevant props, and then stores the config in the enum.

Not entirely sure if that is the best way though, curious to hear your thoughts.

Reply:
I really like this idea!
I can see two options

1. We add the static `init_from_config` method to the FileStore enum impl, store the config in a private struct, and access the config via a new public method on the enum.  We could access the config struct directly but doesn't feel quite right to me (but open to discussion)
2. We create a new FileStorage struct to store the config, `FileStore`, etc. This initially feels better rather than expanding the enum. We could probably call the struct `FileStore` and the enum `Store` as well. 

Open to discussion on implementation on naming here.

Reply:
Not sure if I completely understand storing it in a private struct. It would still be part of the `enum FileStore` than, right?

I'm thinking something like this:

```rust
pub enum FileStore {
   S3(S3Config),
   FS(FSConfig),
}

struct S3Config {
 bucket: String,
 region: String,
  //...etc
}
```

Reply:
Yes, I'm struggling with thinking through the ergonomics of accessing that config on `FileStore` but I probably just need to code it out.

Reply:
So it turns out I was just unfamiliar with the Rust Enum tuple-like members. This all makes sense now. :)